### PR TITLE
[Matrix] use AddonBase.h, LICENSE add, copyright year increase

### PR DESCRIPTION
**Pull request by AlwinEsch**
Started about to change `xbmc_addon_types.h` include to `AddonBase.h`, but found it a bit less and added some other parts :grin: 

Also the on somewhere on other PVR addon requested (by Issue) License is added and is also now shown on README.md, see https://github.com/AlwinEsch/pvr.vuplus/tree/Matrix-change

### CODE REVIEW

File: src/Enigma2.cpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
- *      Copyright (C) 2005-2019 Team XBMC
- *      http://xbmc.org
+ *      Copyright (C) 2005-2020 Team Kodi
+ *      http://kodi.tv
```
Comment (phunkyfish) [inline src/Enigma2.cpp]: This should be `https` everywhere!

**Review by phunkyfish — CHANGES_REQUESTED** [review]

**Comment by phunkyfish** [comment]
Is the AddonBase.h change dependent on a Kodi PR or was it just missing?

File: src/Enigma2.cpp (hunk 1 above)
Comment (AlwinEsch) [reply src/Enigma2.cpp]: Ah OK, becomes changed, was gone down a bit by the round.

**Comment by AlwinEsch** [comment]
With AddonBase.h is not direct related to a Kodi request but to becomes prepared for one soon (with remove of this header). 

See https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_addon_types.h

This already do only a `#include "AddonBase.h"` and no need for me anymore.

File: src/Enigma2.cpp (hunk 1 above)
Comment (AlwinEsch) [reply src/Enigma2.cpp]: Have changed to `https`.

File: .gitignore [hunk 2]
```diff
@@ -1,26 +1,42 @@
...
-.project
-.settings/
+# to prevent add after a "git format-patch VALUE" and "git add ." call
+/*.patch
 
-# VSCode
-.vscode
\ No newline at end of file
+# to prevent add if project code opened by Visual Studio over CMake file
+.vs/
```
Comment (phunkyfish) [inline .gitignore:45]: Why remove the gitignore for VSCode? Bear in mind this is not Visual Studio it's Visual Studio Code!

**Review by phunkyfish — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (AlwinEsch) [reply .gitignore:45]: Sorry, was shot out by c&p from another addons, becomes added back and also look to add on other addons (where missing).

Comment (AlwinEsch) [reply .gitignore:45]: Is back.

**Review by phunkyfish — APPROVED** [review]
Nice, thanks! Will merge shortly.

**Pull request merged**